### ISSUE TITLE
Allow name argument for field, item, and others.

### ISSUE THREAD

**Issue by autarch** [Test-More/Test2-Suite#211]
It'd be nice if I could include that optional final argument for things like `field`. If I have a large complex `hash` block, it can be challenging to understand the meaning of a single failure. I can put comments in, but those don't show up in my failure output.

**Comment by exodist** [comment]
This seems like a viable thing to do, I will look into it.

That said, it sounds like a single comparison may be doing too much, if you are taking the time to comment on every field() declaration then it might be easier, and less verbose to simply put every check that needs such a comment into its own is/ok statement?

This is not a 'no' to the feature, just a suggested alternative for now.

**Comment by autarch** [comment]
> This seems like a viable thing to do, I will look into it.
> 
> That said, it sounds like a single comparison may be doing too much, if you are taking the time to comment on every field() declaration then it might be easier, and less verbose to simply put every check that needs such a comment into its own is/ok statement?

But then I can't take advantage of the really nice failure output, I just get "expected foo but got bar" with no path through the data structure.

**Comment by exodist** [comment]
Next question is how to display the name/comment. In most cases it is probably a sentance, not a single word, which makes fitting it into the table awkward, it is already common for a table to wrap too much. Maybe if there is a comment/name it becomes its own row? But that would require some new features in Term::Table.

**Comment by autarch** [comment]
Yeah, I was thinking either a separate row or after the table entirely.
